src/lib/jsonParseV2/parseSubscription.cpp

```diff
@@ -319,6 +327,12 @@ static std::string parseEntitiesVector(ConnectionInfo* ciP, std::vector<EntID>*
       }
       else if (idPatOpt.given)
       {
+        regex_t regex;
+
+        if (regcomp(&regex, idPatOpt.value.c_str(), 0) != 0)
+        {
+          return badInput(ciP, "invalid regexp in subject entities element idPattern");
+        }
         idPattern = idPatOpt.value;
```

Comment:
With the fix, regfree is called also if regcomp fails.
Not sure what happens if regfree is called with a 'non-regex'.
I would propose to put the call fo regfree **after** returning on invalid regex